sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_policy_token_header.py

```diff
@@ -0,0 +1,164 @@
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import json
+from typing import TypeVar, Any, Union, TYPE_CHECKING, Dict
+from azure.core.pipeline import PipelineRequest
+from azure.core.pipeline.transport import (
+    HttpResponse as LegacyHttpResponse,
+    HttpRequest as LegacyHttpRequest,
+)
+from azure.core.rest import HttpResponse, HttpRequest, AsyncHttpResponse
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+from azure.core.exceptions import HttpResponseError
+
+HTTPResponseType = TypeVar("HTTPResponseType", HttpResponse, LegacyHttpResponse)
+HTTPRequestType = TypeVar("HTTPRequestType", HttpRequest, LegacyHttpRequest)
+
+if TYPE_CHECKING:
+    from .._pipeline_client import ARMPipelineClient
+
+
+def _create_acquire_policy_request(request: PipelineRequest[HTTPRequestType]) -> HttpRequest:
+    """Create a request to acquire a policy token.
+
+    This method creates an HTTP request to the Azure Policy service to acquire
+    a policy token for external policy evaluation. The token will be used in
+    subsequent requests to enable policy evaluation.
+
+    :param request: The pipeline request for which to acquire a policy token
+    :type request: ~azure.core.pipeline.PipelineRequest
+    :return: The HTTP request to acquire policy token
+    :rtype: ~azure.core.rest.HttpRequest
+    :raises ~azure.core.exceptions.HttpResponseError: If subscription ID cannot be extracted from request URL
+    """
+    # try to get subscriptionId from request.http_request.url
+    subscription_id = (
+        request.http_request.url.split("subscriptions/")[1].split("/")[0]
+        if "subscriptions/" in request.http_request.url
+        else None
+    )
+    if not subscription_id:
+        raise HttpResponseError("Failed to get subscriptionId from request url: {}".format(request.http_request.url))
+
+    content = getattr(request.http_request, "content", None)
+    if content and isinstance(content, str):
+        try:
+            content = json.loads(content)
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    body: Dict[str, Any] = {"operation": {"uri": request.http_request.url, "httpMethod": request.http_request.method}}
+    if content:
+        body["operation"]["content"] = content
+    change_reference = request.context.options.pop("change_reference", None)
+    if change_reference:
+        body["changeReference"] = change_reference
+
+    acquire_policy_url = "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/acquirePolicyToken"
+    return HttpRequest(
+        method="POST",
+        url=acquire_policy_url.format(subscriptionId=subscription_id),
+        params={"api-version": "2025-03-01"},
+        headers={"Accept": "application/json", "Content-Type": "application/json", "x-ms-force-sync": "true"},
```

Comment:
I believe we also need the auth header, which can be obtained from the main request.

Reply:
@xiangyan99 A http call may need more info like auth header/ retry header. If we add it directly in this policy, it makes the policy too complicated. So my design is that `_client` shall be configured necessary policies including auth policy for a http call like I show here https://github.com/Azure/azure-sdk-for-python/commit/39108c6b508c7d42ec713d7450f1df3a1309b579#diff-ae510f2fa71df3f0fd26e968097eee953a2eee7dd2f46f2024bdeef961aaeb0a. With this way, this policy does not need too much logic which is needed for a http call. As I described in PR description, I run live test locally to prove the effect of the design.